src/lib.rs

```diff
@@ -411,56 +390,62 @@ pub fn link(config: &CTRConfig, run_args: &Run, verbose: bool) {
     }
 }
 
-/// Read the `RomFS` path from the Cargo manifest. If it's unset, use the default.
-/// The returned boolean is true when the default is used.
-pub fn get_romfs_path(config: &CTRConfig) -> (PathBuf, bool) {
-    let manifest_path = &config.cargo_manifest_path;
-    let manifest_str = std::fs::read_to_string(manifest_path)
-        .unwrap_or_else(|e| panic!("Could not open {}: {e}", manifest_path.display()));
-    let manifest_data: toml::Value =
-        toml::de::from_str(&manifest_str).expect("Could not parse Cargo manifest as TOML");
-
-    // Find the romfs setting and compute the path
-    let mut is_default = false;
-    let romfs_dir_setting = manifest_data
-        .as_table()
-        .and_then(|table| table.get("package"))
-        .and_then(toml::Value::as_table)
-        .and_then(|table| table.get("metadata"))
-        .and_then(toml::Value::as_table)
-        .and_then(|table| table.get("cargo-3ds"))
-        .and_then(toml::Value::as_table)
-        .and_then(|table| table.get("romfs_dir"))
-        .and_then(toml::Value::as_str)
-        .unwrap_or_else(|| {
-            is_default = true;
-            "romfs"
-        });
-    let mut romfs_path = manifest_path.clone();
-    romfs_path.pop(); // Pop Cargo.toml
-    romfs_path.push(romfs_dir_setting);
-
-    (romfs_path, is_default)
-}
-
-#[derive(Default)]
+#[derive(Default, Debug)]
 pub struct CTRConfig {
     name: String,
-    author: String,
+    authors: Vec<String>,
     description: String,
-    icon: String,
-    target_path: PathBuf,
-    cargo_manifest_path: PathBuf,
+    icon_path: Utf8PathBuf,
+    target_path: Utf8PathBuf,
+    romfs_dir: Option<Utf8PathBuf>,
 }
 
 impl CTRConfig {
-    pub fn path_3dsx(&self) -> PathBuf {
+    /// Get the path to the output `.3dsx` file.
+    pub fn path_3dsx(&self) -> Utf8PathBuf {
         self.target_path.with_extension("3dsx")
     }
 
-    pub fn path_smdh(&self) -> PathBuf {
+    /// Get the path to the output `.smdh` file.
+    pub fn path_smdh(&self) -> Utf8PathBuf {
         self.target_path.with_extension("smdh")
     }
+
+    /// Builds the smdh using `smdhtool`.
+    /// This will fail if `smdhtool` is not within the running directory or in a directory found in $PATH
+    pub fn build_smdh(&self, verbose: bool) {
+        let author = if self.authors.is_empty() {
+            String::from("Unspecified Author") // as standard with the devkitPRO toolchain
+        } else {
+            self.authors.join(", ")
```

Comment:
Originally, I believe I made the choice of including only the first author because there was the slight risk that the combined name of all authors might exceed the SMDH limitations (which is a lengthy 128 bytes in UTF-16 https://www.3dbrew.org/wiki/SMDH#Format, but will never be shown in its entirety if it gets this long).

I guess it doesn't really matter either way, since the `authors` field is more-or-less getting abandoned by the Rust community (it's not even shown on crates.io anymore). I'm fine with the change.

Reply:
Ah, hmm... from looking at the `smdhtool` implementation, it seems like anything longer will just get truncated to `0x40` characters anyway: https://github.com/devkitPro/3dstools/blob/master/src/smdhtool.cpp#L423-L429

We could add a length check of our own and panic if the joined string is too long, but it would require calculating the UTF-16 length so I think just letting it get truncated seems fine. Once #58 is implemented users will also be able to override the author string to their liking without needing to modify `Cargo.toml`'s `package.authors` field.